src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -56,115 +57,181 @@ private void RegisterAnalyzers(CompilationAnalysisContext context)
-            if (returnType == "System.Int")
-                return returnType + "32";
-            return returnType;
+            
+            return null;
         }
 
-        public void FindActivities(SyntaxNodeAnalysisContext context)
+        private string GetQualifiedTypeName(ITypeSymbol typeInfo)
         {
-            var attributeExpression = context.Node as AttributeSyntax;
-            if (attributeExpression != null && attributeExpression.ChildNodes().First().ToString() == "ActivityTrigger")
+            if (typeInfo != null)
             {
-                if (SyntaxNodeUtils.TryGetFunctionAttribute(attributeExpression, out SyntaxNode functionAttribute))
+                if (typeInfo is INamedTypeSymbol)
                 {
-                    if (SyntaxNodeUtils.TryGetFunctionName(functionAttribute, out SyntaxNode attributeArgument))
+                    var tupleUnderlyingType = ((INamedTypeSymbol)typeInfo).TupleUnderlyingType;
+                    if (tupleUnderlyingType != null)
                     {
-                        var functionName = attributeArgument.ToString().Trim('"');
-                        if (SyntaxNodeUtils.TryGetParameterNodeNextToAttribute(attributeExpression, context, out SyntaxNode inputTypeNode))
-                        {
-                            ITypeSymbol inputType = context.SemanticModel.GetTypeInfo(inputTypeNode).Type;
-                            if (inputType.ToString().Equals("Microsoft.Azure.WebJobs.Extensions.DurableTask.IDurableActivityContext") || inputType.ToString().Equals("Microsoft.Azure.WebJobs.DurableActivityContext"))
-                            {
-                                if (!TryGetInputTypeFromDurableContextCall(out inputType, context, attributeExpression))
-                                {
-                                    return;
-                                }
-                            }
-
-                            if (SyntaxNodeUtils.TryGetReturnType(attributeExpression, context, out ITypeSymbol returnType))
-                            {
-                                availableFunctions.Add(new ActivityFunctionDefinition
-                                {
-                                    FunctionName = functionName,
-                                    InputType = GetQualifiedTypeName(inputType),
-                                    ReturnType = GetQualifiedTypeName(returnType)
-                                });
-                            }
-                        }
+                        return $"System.Tuple<{string.Join(", ", tupleUnderlyingType.TypeArguments.Select(x => x.ToString()))}>";
                     }
+                    return typeInfo.ToString();
                 }
+
+                return typeInfo.ContainingNamespace?.ToString() + "." + typeInfo.Name?.ToString();
             }
+
+            return "";
+        }
+
+        public void FindActivities(SyntaxNodeAnalysisContext context)
```

Comment:
Consider changing to the singular `FindActivity()`, since it doesn't look like it ever adds more than one. An even more descriptive name could be "AddActivityIfExists()`, but I am not sold on that.